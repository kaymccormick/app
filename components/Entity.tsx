import { EntityProps } from './proptypes';
import React from 'react';

export default (props: EntityProps) => <div/>;
