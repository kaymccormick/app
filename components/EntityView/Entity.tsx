import React from 'react';
/*import Popover from 'react-simple-popover';*/
import {Types} from '../../src/types';
import {DragSource, DragSourceConnector, DragSourceMonitor} from 'react-dnd';
import * as Components from './';
import {EntityDragDropItem, EntityProps,EntityMethodsSectionProps,EntityAttributesSectionProps} from "../types";

const defaultWidth = 100;
const defaultHeight = 400;

const Sections = {
    attributes: { render: (props: EntityAttributesSectionProps) => <Components.EntityAttributesSection index={props.index||-1} key={props.section} {...props}/> },
    methods: { render: (props: EntityMethodsSectionProps) => <Components.EntityMethodsSection index={props.index} key={props.section} {...props}/> },
};

const entitySource = {
    beginDrag(props: EntityProps): EntityDragDropItem {
        // const r = this.myRef.current.getBoundingClientRect();
        const item = { modelKey: props.modelKey || '',
        // clientRect: { x: r.x, y: r.y, width: r.width, height: r.height },
        };
        return item;
    },
    endDrag(props: EntityProps, monitor: DragSourceMonitor, component: React.RefObject<any>) {
        if (!monitor.didDrop()) {
            return;
        }
        const item = monitor.getItem();
        console.log(item);
    },
};


function collect(connect: DragSourceConnector, monitor: DragSourceMonitor) {
    return {
        connectDragSource: connect.dragSource(),
        isDragging: monitor.isDragging(),

    };
}

class Entity extends React.Component<EntityProps> {
private target: any;
    private myRef: React.RefObject<any>;
    // @ts-ignore
    public handleOnClick: any;
    public constructor(props: EntityProps) {
        super(props);
        console.log(props);
        this.myRef = React.createRef();
        this.handleOnClick = this._handleOnClick.bind(this);
    }

    componentDidMount() {
        const rect = this.myRef.current.getBoundingClientRect();
        console.log(rect);
    }
    public _handleOnClick(e: MouseEvent) {
      e.preventDefault();
      console.log('onclick');
      if(this.props.changeSelection) {
        this.props.changeSelection(this.props.index);
        }
        }

      public render() {
        const draggableEntityStyle: React.CSSProperties = {
              position: "absolute",
              left: (this.props.x || 50),
              top: (this.props.y || 100),
              right: ((this.props.x || 50) + (this.props.width || defaultWidth)),
              bottom: ((this.props.y || 100) + (this.props.height || defaultHeight)),
              };

        const {isDragging, connectDragSource} = this.props;
        return connectDragSource(
         <div className="entity__outerGrid">
           <div className="entityView__draggableEntity"
                ref={this.myRef}
                style={draggableEntityStyle}>
            <div className="entityView__entityContainer">
              <div className="entityView__entity" onClick={this.handleOnClick}>
              <div className="entity__displayName">
                {this.props.editMode ? <input value={this.props.entity !== undefined ?
                                     this.props.entity.displayName: ''}/> :
                  <span>{this.props.entity !== undefined ? this.props.entity.displayName: ''}</span>}
                </div>
                <div className="entity__header">Description</div>
                <div className="entity__description">hello</div>
            {Object.keys(Sections).map(section => Sections[section].render(
	    { index: this.props.index,
            entity: this.props.entity,
	    section }))}
            <div><a href="#" onClick={(e) => { e.preventDefault(); if(this.props.updateEntity) { this.props.updateEntity(); } }}>Update</a></div>
            </div></div><div style={{width: '1px'}} ref={(node) => { this.target = node; }}/></div></div>);
    }
}

export default DragSource(Types.ENTITY, entitySource, collect)(Entity);
