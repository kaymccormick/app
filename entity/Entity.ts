import {Entity as OrmEntity, Column, PrimaryGeneratedColumn } from "typeorm";

@OrmEntity()
export default class Entity {
  @PrimaryGeneratedColumn()
    public  id: number;
  @Column()
  public   name: string;
  @Column()
  public   dynamic: boolean;
  @Column()
  public  objectId: number;

  public constructor(entityid: number, name: string, dynamic: boolean, objectId: number) {
      this.entityid = entityid;
      this.name = name;
      this.dynamic = dynamic;
      this.objectId = objectId;
  }
}

