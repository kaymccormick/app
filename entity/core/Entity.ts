import { Entity as OrmEntity, Column, PrimaryGeneratedColumn, OneToMany, OneToOne, ManyToOne } from "typeorm";
import EntityAttribute from './EntityAttribute';

@OrmEntity()
export default class Entity {
    @PrimaryGeneratedColumn()
    public id?: number;

    @Column({nullable: true})
    public commonName?: string;

    @Column({nullable: true})
    public displayName?: string;

    @OneToMany(type => EntityAttribute, att => att.entity)
    public attributes?: EntityAttribute[]
}
