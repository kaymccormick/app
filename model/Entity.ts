import EntityAttribute from './EntityAttribute';

export default class Entity {
    public id?: number;

    public commonName?: string;

    public displayName?: string;

    public attributes?: EntityAttribute[];

public toJS(): { id?: number, commonName?: string, displayName?: string,
attributes: {}[] } {
return { id: this.id,
commonName: this.commonName,
displayName: this.displayName,
attributes: this.attributes !== undefined ? this.attributes.map(a => a.toJS()) : [],
}
}
}


