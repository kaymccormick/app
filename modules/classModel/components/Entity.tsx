import React from 'react';
import { Pojo } from 'classModel';
//@ts-ignore
import { Rect } from 'react-konva';

export interface Props {
}
export default class Entity extends React.Component<Props> {
    render() {
         return <Rect x={20} y={20} width={50} height={50}/>;
    }
}
