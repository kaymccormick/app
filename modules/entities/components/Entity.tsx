import React from 'react';
import {EntityPojo} from '../types';
import * as Components from './';

export interface EntityProps {
    entity?: EntityPojo;
}

export class Entity extends React.Component<EntityProps> {
    private ref: React.RefObject<any>;

constructor(props:EntityProps) {
super(props);
this.ref = React.createRef();
}
    public componentDidMount() {
    if(this.ref.current) {
        const rect = this.ref.current.getBoundingClientRect();
        console.log(rect);
        }
    }

    public render() {
        const outerStyles:React.CSSProperties = {
            margin:'.5rem',
            borderCollapse: 'collapse',
            gridRowGap:'0px',
            gridColumnGap:'0px',
            border: '1px solid black',
            display: 'inline-grid',
            gridTemplateRows: 'auto-fit 1fr'};

        const nameStyles: React.CSSProperties = {
        paddingLeft: '5px',
            paddingRight: '5px',
            paddingTop: '3px',
            paddingBottom:'3px',
            borderBottom: '1px solid black',
            gridColumn:'span 2'};

        const name = this.props.entity ? this.props.entity.name : '';
        return <div ref={this.ref} style={outerStyles}>
            <div style={nameStyles}>{name}</div>
            {this.props.entity && this.props.entity.columns
            ? this.props.entity.columns.map((column: any) => column ? <Components.EntityColumn column={column}/> :null) : null}
        </div>;
    }
}
