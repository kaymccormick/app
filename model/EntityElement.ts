import Entity from './Entity';

export default class EntityElement {
    public id?: number;
    public commonName?: string;
    public displayName?: string;

    public entity?: Entity;
}
