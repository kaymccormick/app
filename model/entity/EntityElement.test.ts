import EntityElement from './EntityElement';
import Entity from './Entity';

test('1', () => {
    const ee = new EntityElement();
    ee.entity = new Entity(1);
    expect(ee.toJS()).toMatchSnapshot();
    const ee2 = ee.copy();
    expect(ee2).toEqual(ee);
    expect(ee2.toJS()).toMatchSnapshot();
    });
    
