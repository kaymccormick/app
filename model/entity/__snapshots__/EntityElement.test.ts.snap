// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`1 1`] = `
Object {
  "commonName": undefined,
  "displayName": undefined,
  "entity": Object {
    "attributes": Object {
      "sectionContents": Array [],
    },
    "commonName": undefined,
    "displayName": undefined,
    "objectid": 1,
  },
  "isNew": undefined,
  "key": undefined,
  "objectid": undefined,
  "owner": undefined,
  "tenant": undefined,
}
`;

exports[`1 2`] = `
Object {
  "commonName": undefined,
  "displayName": undefined,
  "isNew": undefined,
  "key": undefined,
  "objectid": undefined,
  "owner": undefined,
  "tenant": undefined,
}
`;
